src/components/document-metadata-modal.js

```diff
@@ -25,9 +29,121 @@ const logger = new Logger(import.meta.url);
+
+  const strategyOptions = [
+    {
+      label: documentToClone.roomId
+        ? t('cloningStrategy_cloneWithinArea_fromRoom')
+        : t('cloningStrategy_cloneWithinArea_fromPublicArea'),
+      value: CLONING_STRATEGY.cloneWithinArea
+    }
+  ];
+
+  if (clientConfig.areRoomsEnabled) {
+    strategyOptions.push({
+      label: documentToClone.roomId
+        ? t('cloningStrategy_crossCloneIntoRoom_fromRoom')
+        : t('cloningStrategy_crossCloneIntoRoom_fromPublicArea'),
+      value: CLONING_STRATEGY.crossCloneIntoRoom
+    });
+  }
+
+  if (documentToClone.roomId) {
+    strategyOptions.push({
+      label: t('cloningStrategy_crossCloneIntoPublicArea'),
+      value: CLONING_STRATEGY.crossCloneIntoPublicArea
+    });
+  }
+
+  return { strategyOptions, roomOptions };
+}
+
+function determineActualTemplateDocumentId({ mode, documentToClone, useTemplateDocument, defaultTemplateDocumentId }) {
+  switch (mode) {
+    case DOCUMENT_METADATA_MODAL_MODE.clone:
+      return documentToClone._id;
+    case DOCUMENT_METADATA_MODAL_MODE.create:
+      return useTemplateDocument ? defaultTemplateDocumentId : null;
+    case DOCUMENT_METADATA_MODAL_MODE.update:
+      return null;
+    default:
+      throw new Error(`Invalid document metadata modal mode: '${mode}'`);
+  }
+}
+
+function determineTargetRoomId({ mode, initialDocumentMetadata, documentToClone, cloningStrategy, cloningTargetRoomId }) {
+  switch (mode) {
+    case DOCUMENT_METADATA_MODAL_MODE.clone:
+      switch (cloningStrategy) {
+        case CLONING_STRATEGY.cloneWithinArea:
+          return documentToClone.roomId || null;
+        case CLONING_STRATEGY.crossCloneIntoRoom:
+          return cloningTargetRoomId;
+        case CLONING_STRATEGY.crossCloneIntoPublicArea:
+          return null;
+        default:
+          throw new Error(`Invalid cloning strategy: '${cloningStrategy}'`);
+      }
+    case DOCUMENT_METADATA_MODAL_MODE.create:
+      return initialDocumentMetadata.roomId || null;
+    case DOCUMENT_METADATA_MODAL_MODE.update:
+      return null; // Not used for metadata updates
+    default:
+      throw new Error(`Invalid document metadata modal mode: '${mode}'`);
+  }
+}
+
+function getDialogTitle(mode, t) {
+  switch (mode) {
+    case DOCUMENT_METADATA_MODAL_MODE.clone:
+      return t('cloneDocument');
+    case DOCUMENT_METADATA_MODAL_MODE.create:
+      return t('newDocument');
+    case DOCUMENT_METADATA_MODAL_MODE.update:
+      return t('editDocument');
+    default:
+      throw new Error(`Invalid document metadata modal mode: '${mode}'`);
+  }
+}
+
+function getDialogOkButtonText(mode, t) {
+  switch (mode) {
+    case DOCUMENT_METADATA_MODAL_MODE.clone:
+      return t('common:clone');
+    case DOCUMENT_METADATA_MODAL_MODE.create:
+      return t('common:create');
+    case DOCUMENT_METADATA_MODAL_MODE.update:
+      return t('common:save');
+    default:
+      throw new Error(`Invalid document metadata modal mode: '${mode}'`);
+  }
+}
+
```

Comment:
Maybe we could move all these helper functions to a `document-metadata-modal-utils.js` module? It's alot of code in here.

Reply:
Yeah, good idea, it was bothering me, too. I'll do that!

Reply:
Done.

Reply:
Nice! 👍